README.txt
Minetest mod "Hunger"
=====================
Version: 1.1.2

(c) Copyright BlockMen (2015)


About this mod:
~~~~~~~~~~~~~~~
This mod adds hunger mechanics to Minetest, which are based on player actions and on time.
Also it changes the eating in Minetest, e.g. an Apple does not restore Health, but it rises your saturation.
Example: 1 apple fills up the hunger bar by 1 "bread" (statbar symbol).
Although the statbar show 20 hunger points (10 breads) on the HUD you can fill it up to 30 points.

By default it supports a lot of food already (see full list below) and food that for registered via the API.
For more information how to register more food see API.txt

Information:
This mod depends on the "Better HUD" mod (https://github.com/BlockMen/hud) to provide information about your current saturation.


For Modders:
~~~~~~~~~~~~
This mod alters the behavior of minetest.item_eat().
All callbacks that are registered via minetest.register_on_item_eat() are called AFTER this mod actions, so the itemstack
will have changed already when callbacks are called. You can get the original itemstack as 6th parameter of your function then.

License:
~~~~~~~~
(c) Copyright BlockMen (2015)


Code:
Licensed under the GNU LGPL version 2.1 or higher.
You can redistribute it and/or modify it under 
the terms of the GNU Lesser General Public License 
as published by the Free Software Foundation;

You should have received a copy of the GNU Lesser General Public
License along with this library; if not, write to the Free Software
Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA

See LICENSE.txt and http://www.gnu.org/licenses/lgpl-2.1.txt


Textures:
hunger_statbar_poisen.png - BlockMen (CC-BY 3.0)

Sounds:
hunger_eat.ogg - BlockMen (CC-BY 3.0)


Github:
~~~~~~~
https://github.com/BlockMen/hunger

Forum:
~~~~~~
-

Changelog:
~~~~~~~~~~
see changelog.txt


Dependencies:
~~~~~~~~~~~~~
- Default
- Farming
- Better HUD (https://github.com/BlockMen/hud)


Supported food/mods:
~~~~~~~~~~~~~~~~~~~~
- Apples (default)
- Animalmaterials (mobf modpack)
- Bread (default)
- Bushes
- bushes_classic
- Creatures
- Dwarves (beer and such)
- Docfarming
- Fishing
- Farming plus
- Farming (default and Tenplus1's fork)
- Food
- fruit
- Glooptest
- JKMod
- kpgmobs
- Mobfcooking
- Mooretrees
- Mtfoods
- mushroom
- mush45
- Seaplants (sea)
- Simple mobs


changelog.txt
1.0
---
- Partly rewritten (based on Better HUD hunger mechanics)
- Added API to register food
- Added eating sounds
- Hungerbar image changes when poisend

1.1
---
- Fixed healing after death
- Fixed healing while drowning
- Fixed crashed caused by Pipeworks mod
- Added wrapper for minetest.item_eat(). see Readme.txt for more informations
- Updated HUD-API usage
- Added beans of Farming Redo

1.1.1
-----
- Fixed unhandled exception (reported by Martin_Devil)
- Added support for mtg mushrooms

1.1.2
-----
- Fixed crash on invalid values given


init.lua
hunger = {}
hunger.food = {}

HUNGER_TICK = 800			-- time in seconds after that 1 hunger point is taken
HUNGER_HEALTH_TICK = 4		-- time in seconds after player gets healed/damaged
HUNGER_MOVE_TICK = 0.5		-- time in seconds after the movement is checked

HUNGER_EXHAUST_DIG = 3		-- exhaustion increased this value after digged node
HUNGER_EXHAUST_PLACE = 1	-- exhaustion increased this value after placed
HUNGER_EXHAUST_MOVE = 1.5	-- exhaustion increased this value if player movement detected
HUNGER_EXHAUST_LVL = 160	-- at what exhaustion player saturation gets lowered

HUNGER_HEAL = 1				-- number of HP player gets healed after HUNGER_HEALTH_TICK
HUNGER_HEAL_LVL = 15		-- lower level of saturation needed to get healed
HUNGER_STARVE = 1			-- number of HP player gets damaged by hunger after HUNGER_HEALTH_TICK
HUNGER_STARVE_LVL = 3		-- level of staturation that causes starving

HUNGER_MAX = 30				-- maximum level of saturation


local modpath = minetest.get_modpath("hunger")
dofile(modpath .. "/functions.lua")
dofile(modpath .. "/food.lua")
dofile(modpath .. "/legacy.lua")


-- Callbacks
if minetest.setting_getbool("enable_damage") then
    minetest.register_on_joinplayer(function(player)
	local inv = player:get_inventory()
	inv:set_size("hunger", 1)

	local name = player:get_player_name()
	hunger[name] = {}
	hunger[name].lvl = hunger.read(player)
	hunger[name].exhaus = 0
	local lvl = hunger[name].lvl
	if lvl > 20 then
		lvl = 20
	end
	minetest.after(0.8, function()
		hud.swap_statbar(player, "hunger", "air")
		hud.change_item(player, "hunger", {number = lvl, max = 20})
	end)
    end)

    -- for exhaustion
    minetest.register_on_placenode(hunger.handle_node_actions)
    minetest.register_on_dignode(hunger.handle_node_actions)
    minetest.register_on_respawnplayer(function(player)
	hunger.update_hunger(player, 20)
    end)
end


functions.lua
-- read/write
function hunger.read(player)
	local inv = player:get_inventory()
	if not inv then
		return nil
	end
	local hgp = inv:get_stack("hunger", 1):get_count()
	if hgp == 0 then
		hgp = 21
		inv:set_stack("hunger", 1, ItemStack({name = ":", count = hgp}))
	else
		hgp = hgp
	end
	if tonumber(hgp) > HUNGER_MAX + 1 then
		hgp = HUNGER_MAX + 1
	end
	return hgp - 1
end

function hunger.save(player)
	local inv = player:get_inventory()
	local name = player:get_player_name()
	local value = hunger[name].lvl
	if not inv or not value then
		return nil
	end
	if value > HUNGER_MAX then
		value = HUNGER_MAX
	end
	if value < 0 then
		value = 0
	end
	inv:set_stack("hunger", 1, ItemStack({name = ":", count = value + 1}))
	return true
end

function hunger.update_hunger(player, new_lvl)
	local name = player:get_player_name() or nil
	if not name then
		return false
	end
	if minetest.setting_getbool("enable_damage") == false then
		hunger[name] = 20
		return
	end
	local lvl = hunger[name].lvl
	if new_lvl then
		 lvl = new_lvl
	end
	if lvl > HUNGER_MAX then
		lvl = HUNGER_MAX
	end
	hunger[name].lvl = lvl
	if lvl > 20 then
		lvl = 20
	end
	hud.change_item(player, "hunger", {number = lvl})
	hunger.save(player)
end
local update_hunger = hunger.update_hunger

-- player-action based hunger changes
function hunger.handle_node_actions(pos, oldnode, player, ext)
	if not player or not player:is_player() then
		return
	end
	local name = player:get_player_name()
	if not name or not hunger[name] then
		return
	end

	local exhaus = hunger[name].exhaus
	if not exhaus then
		hunger[name].exhaus = 0
		--return
	end

	local new = HUNGER_EXHAUST_PLACE

	-- placenode event
	if not ext then
		new = HUNGER_EXHAUST_DIG
	end

	-- assume its send by action_timer(globalstep)
	if not pos and not oldnode then
		new = HUNGER_EXHAUST_MOVE
	end

	exhaus = exhaus + new

	if exhaus > HUNGER_EXHAUST_LVL then
		exhaus = 0
		local h = tonumber(hunger[name].lvl)
		if h > 0 then
			update_hunger(player, h - 1)
		end
	end

	hunger[name].exhaus = exhaus
end


-- Time based hunger functions
local hunger_timer = 0
local health_timer = 0
local action_timer = 0

local function hunger_globaltimer(dtime)
	hunger_timer = hunger_timer + dtime
	health_timer = health_timer + dtime
	action_timer = action_timer + dtime

	if action_timer > HUNGER_MOVE_TICK then
		for _,player in ipairs(minetest.get_connected_players()) do
			local controls = player:get_player_control()
			-- Determine if the player is walking
			if controls.up or controls.down or controls.left or controls.right then
				hunger.handle_node_actions(nil, nil, player)
			end
		end
		action_timer = 0
	end

	-- lower saturation by 1 point after <HUNGER_TICK> second(s)
	if hunger_timer > HUNGER_TICK then
		for _,player in ipairs(minetest.get_connected_players()) do
			local name = player:get_player_name()
			local tab = hunger[name]
			if tab then
				local hunger = tab.lvl
				if hunger > 0 then
					update_hunger(player, hunger - 1)
				end
			end
		end
		hunger_timer = 0
	end

	-- heal or damage player, depending on saturation
	if health_timer > HUNGER_HEALTH_TICK then
		for _,player in ipairs(minetest.get_connected_players()) do
			local name = player:get_player_name()
			local tab = hunger[name]
			if tab then
				local air = player:get_breath() or 0
				local hp = player:get_hp()

				-- heal player by 1 hp if not dead and saturation is > 15 (of 30) player is not drowning
				if tonumber(tab.lvl) > HUNGER_HEAL_LVL and hp > 0 and air > 0 then
					player:set_hp(hp + HUNGER_HEAL)
				end

				-- or damage player by 1 hp if saturation is < 2 (of 30)
				if tonumber(tab.lvl) < HUNGER_STARVE_LVL then
					player:set_hp(hp - HUNGER_STARVE)
				end
			end
		end

		health_timer = 0
	end
end

if minetest.setting_getbool("enable_damage") then
	minetest.register_globalstep(hunger_globaltimer)
end


-- food functions
local food = hunger.food

function hunger.register_food(name, hunger_change, replace_with_item, poisen, heal, sound)
	food[name] = {}
	food[name].saturation = hunger_change	-- hunger points added
	food[name].replace = replace_with_item	-- what item is given back after eating
	food[name].poisen = poisen				-- time its poisening
	food[name].healing = heal				-- amount of HP
	food[name].sound = sound				-- special sound that is played when eating
end

-- Poison player
local function poisenp(tick, time, time_left, player)
	time_left = time_left + tick
	if time_left < time then
		minetest.after(tick, poisenp, tick, time, time_left, player)
	else
		hud.change_item(player, "hunger", {text = "hud_hunger_fg.png"})
	end
	local hp = player:get_hp() -1 or 0
	if hp > 0 then
		player:set_hp(hp)
	end
end

-- wrapper for minetest.item_eat (this way we make sure other mods can't break this one)
local org_eat = core.do_item_eat
core.do_item_eat = function(hp_change, replace_with_item, itemstack, user, pointed_thing)
	local old_itemstack = itemstack
	itemstack = hunger.eat(hp_change, replace_with_item, itemstack, user, pointed_thing)
	for _, callback in pairs(core.registered_on_item_eats) do
		local result = callback(hp_change, replace_with_item, itemstack, user, pointed_thing, old_itemstack)
		if result then
			return result
		end
	end
	return itemstack
end

function hunger.eat(hp_change, replace_with_item, itemstack, user, pointed_thing)
	local item = itemstack:get_name()
	local def = food[item]
	if not def then
		def = {}
		if type(hp_change) ~= "number" then
			hp_change = 1
			core.log("error", "Wrong on_use() definition for item '" .. item .. "'")
		end
		def.saturation = hp_change * 1.3
		def.replace = replace_with_item
	end
	local func = hunger.item_eat(def.saturation, def.replace, def.poisen, def.healing, def.sound)
	return func(itemstack, user, pointed_thing)
end

function hunger.item_eat(hunger_change, replace_with_item, poisen, heal, sound)
    return function(itemstack, user, pointed_thing)
	if itemstack:take_item() ~= nil and user ~= nil then
		local name = user:get_player_name()
		if not hunger[name] then
			return itemstack
		end
		local sat = tonumber(hunger[name].lvl or 0)
		local hp = user:get_hp()
		-- Saturation
		if sat < HUNGER_MAX and hunger_change then
			sat = sat + hunger_change
			hunger.update_hunger(user, sat)
		end
		-- Healing
		if hp < 20 and heal then
			hp = hp + heal
			if hp > 20 then
				hp = 20
			end
			user:set_hp(hp)
		end
		-- Poison
		if poisen then
			hud.change_item(user, "hunger", {text = "hunger_statbar_poisen.png"})
			poisenp(1.0, poisen, 0, user)
		end

		-- eating sound
		if not sound then
			sound = "hunger_eat"
		end
		minetest.sound_play(sound, {to_player = name, gain = 0.7})

		if replace_with_item then
			if itemstack:is_empty() then
				itemstack:add_item(replace_with_item)
			else
				local inv = user:get_inventory()
				if inv:room_for_item("main", {name=replace_with_item}) then
					inv:add_item("main", replace_with_item)
				else
					local pos = user:getpos()
					pos.y = math.floor(pos.y + 0.5)
					core.add_item(pos, replace_with_item)
				end
			end
		end
	end

	return itemstack
    end
end


food.lua
local register_food = hunger.register_food

register_food("default:apple", 2)

if minetest.get_modpath("farming") ~= nil then
	register_food("farming:bread", 4)
end

if minetest.get_modpath("mobs") ~= nil then
	if mobs.mod ~= nil and mobs.mod == "redo" then
		register_food("mobs:cheese", 4)
		register_food("mobs:meat", 8)
		register_food("mobs:meat_raw", 4)
		register_food("mobs:rat_cooked", 4)
		register_food("mobs:honey", 2)
		register_food("mobs:pork_raw", 3, "", 3)
		register_food("mobs:pork_cooked", 8)
		register_food("mobs:chicken_cooked", 6)
		register_food("mobs:chicken_raw", 2, "", 3)
		register_food("mobs:chicken_egg_fried", 2)
		if minetest.get_modpath("bucket") then 
			register_food("mobs:bucket_milk", 3, "bucket:bucket_empty")
		end
	else
		register_food("mobs:meat", 6)
		register_food("mobs:meat_raw", 3)
		register_food("mobs:rat_cooked", 5)
	end
end

if minetest.get_modpath("moretrees") ~= nil then
	register_food("moretrees:coconut_milk", 1)
	register_food("moretrees:raw_coconut", 2)
	register_food("moretrees:acorn_muffin", 3)
	register_food("moretrees:spruce_nuts", 1)
	register_food("moretrees:pine_nuts", 1)
	register_food("moretrees:fir_nuts", 1)
end

if minetest.get_modpath("dwarves") ~= nil then
	register_food("dwarves:beer", 2)
	register_food("dwarves:apple_cider", 1)
	register_food("dwarves:midus", 2)
	register_food("dwarves:tequila", 2)
	register_food("dwarves:tequila_with_lime", 2)
	register_food("dwarves:sake", 2)
end

if minetest.get_modpath("animalmaterials") ~= nil then
	register_food("animalmaterials:milk", 2)
	register_food("animalmaterials:meat_raw", 3)
	register_food("animalmaterials:meat_pork", 3)
	register_food("animalmaterials:meat_beef", 3)
	register_food("animalmaterials:meat_chicken", 3)
	register_food("animalmaterials:meat_lamb", 3)
	register_food("animalmaterials:meat_venison", 3)
	register_food("animalmaterials:meat_undead", 3, "", 3)
	register_food("animalmaterials:meat_toxic", 3, "", 5)
	register_food("animalmaterials:meat_ostrich", 3)
	register_food("animalmaterials:fish_bluewhite", 2)
	register_food("animalmaterials:fish_clownfish", 2)
end

if minetest.get_modpath("fishing") ~= nil then
	register_food("fishing:fish_raw", 2)
	register_food("fishing:fish_cooked", 5)
	register_food("fishing:sushi", 6)
	register_food("fishing:shark", 4)
	register_food("fishing:shark_cooked", 8)
	register_food("fishing:pike", 4)
	register_food("fishing:pike_cooked", 8)
end

if minetest.get_modpath("glooptest") ~= nil then
	register_food("glooptest:kalite_lump", 1)
end

if minetest.get_modpath("bushes") ~= nil then
	register_food("bushes:sugar", 1)
	register_food("bushes:strawberry", 2)
	register_food("bushes:berry_pie_raw", 3)
	register_food("bushes:berry_pie_cooked", 4)
	register_food("bushes:basket_pies", 15)
end

if minetest.get_modpath("bushes_classic") then
	-- bushes_classic mod, as found in the plantlife modpack
	local berries = {
	    "strawberry",
		"blackberry",
		"blueberry",
		"raspberry",
		"gooseberry",
		"mixed_berry"}
	for _, berry in ipairs(berries) do
		if berry ~= "mixed_berry" then
			register_food("bushes:"..berry, 1)
		end
		register_food("bushes:"..berry.."_pie_raw", 2)
		register_food("bushes:"..berry.."_pie_cooked", 5)
		register_food("bushes:basket_"..berry, 15)
	end
end

if minetest.get_modpath("mushroom") ~= nil then
	register_food("mushroom:brown", 1)
	register_food("mushroom:red", 1, "", 3)
	-- mushroom potions: red = strong poison, brown = light restorative
	if minetest.get_modpath("vessels") then
		register_food("mushroom:brown_essence", 1, "vessels:glass_bottle", nil, 4)
		register_food("mushroom:poison", 1, "vessels:glass_bottle", 10)
	end
end

if minetest.get_modpath("docfarming") ~= nil then
	register_food("docfarming:carrot", 3)
	register_food("docfarming:cucumber", 2)
	register_food("docfarming:corn", 3)
	register_food("docfarming:potato", 4)
	register_food("docfarming:bakedpotato", 5)
	register_food("docfarming:raspberry", 3)
end

if minetest.get_modpath("farming_plus") ~= nil then
	register_food("farming_plus:carrot_item", 3)
	register_food("farming_plus:banana", 2)
	register_food("farming_plus:orange_item", 2)
	register_food("farming:pumpkin_bread", 4)
	register_food("farming_plus:strawberry_item", 2)
	register_food("farming_plus:tomato_item", 2)
	register_food("farming_plus:potato_item", 4)
	register_food("farming_plus:rhubarb_item", 2)
end

if minetest.get_modpath("mtfoods") ~= nil then
	register_food("mtfoods:dandelion_milk", 1)
	register_food("mtfoods:sugar", 1)
	register_food("mtfoods:short_bread", 4)
	register_food("mtfoods:cream", 1)
	register_food("mtfoods:chocolate", 2)
	register_food("mtfoods:cupcake", 2)
	register_food("mtfoods:strawberry_shortcake", 2)
	register_food("mtfoods:cake", 3)
	register_food("mtfoods:chocolate_cake", 3)
	register_food("mtfoods:carrot_cake", 3)
	register_food("mtfoods:pie_crust", 3)
	register_food("mtfoods:apple_pie", 3)
	register_food("mtfoods:rhubarb_pie", 2)
	register_food("mtfoods:banana_pie", 3)
	register_food("mtfoods:pumpkin_pie", 3)
	register_food("mtfoods:cookies", 2)
	register_food("mtfoods:mlt_burger", 5)
	register_food("mtfoods:potato_slices", 2)
	register_food("mtfoods:potato_chips", 3)
	--mtfoods:medicine
	register_food("mtfoods:casserole", 3)
	register_food("mtfoods:glass_flute", 2)
	register_food("mtfoods:orange_juice", 2)
	register_food("mtfoods:apple_juice", 2)
	register_food("mtfoods:apple_cider", 2)
	register_food("mtfoods:cider_rack", 2)
end

if minetest.get_modpath("fruit") ~= nil then
	register_food("fruit:apple", 2)
	register_food("fruit:pear", 2)
	register_food("fruit:bananna", 3)
	register_food("fruit:orange", 2)
end

if minetest.get_modpath("mush45") ~= nil then
	register_food("mush45:meal", 4)
end

if minetest.get_modpath("seaplants") ~= nil then
	register_food("seaplants:kelpgreen", 1)
	register_food("seaplants:kelpbrown", 1)
	register_food("seaplants:seagrassgreen", 1)
	register_food("seaplants:seagrassred", 1)
	register_food("seaplants:seasaladmix", 6)
	register_food("seaplants:kelpgreensalad", 1)
	register_food("seaplants:kelpbrownsalad", 1)
	register_food("seaplants:seagrassgreensalad", 1)
	register_food("seaplants:seagrassgreensalad", 1)
end

if minetest.get_modpath("mobfcooking") ~= nil then
	register_food("mobfcooking:cooked_pork", 6)
	register_food("mobfcooking:cooked_ostrich", 6)
	register_food("mobfcooking:cooked_beef", 6)
	register_food("mobfcooking:cooked_chicken", 6)
	register_food("mobfcooking:cooked_lamb", 6)
	register_food("mobfcooking:cooked_venison", 6)
	register_food("mobfcooking:cooked_fish", 6)
end

if minetest.get_modpath("creatures") ~= nil then
	register_food("creatures:meat", 6)
	register_food("creatures:flesh", 3)
	register_food("creatures:rotten_flesh", 3, "", 3)
end

if minetest.get_modpath("ethereal") then
	register_food("ethereal:strawberry", 1)
	register_food("ethereal:banana", 4)
	register_food("ethereal:pine_nuts", 1)
	register_food("ethereal:bamboo_sprout", 0, "", 3)
	register_food("ethereal:fern_tubers", 1)
	register_food("ethereal:banana_bread", 7)
	register_food("ethereal:mushroom_plant", 2)
	register_food("ethereal:coconut_slice", 2)
	register_food("ethereal:golden_apple", 4, "", nil, 10)
	register_food("ethereal:wild_onion_plant", 2)
	register_food("ethereal:mushroom_soup", 4, "ethereal:bowl")
	register_food("ethereal:mushroom_soup_cooked", 6, "ethereal:bowl")
	register_food("ethereal:hearty_stew", 6, "ethereal:bowl", 3)
	register_food("ethereal:hearty_stew_cooked", 10, "ethereal:bowl")
	if minetest.get_modpath("bucket") then
		register_food("ethereal:bucket_cactus", 2, "bucket:bucket_empty")
	end
	register_food("ethereal:fish_raw", 2)
	register_food("ethereal:fish_cooked", 5)
	register_food("ethereal:seaweed", 1)
	register_food("ethereal:yellowleaves", 1, "", nil, 1)
	register_food("ethereal:sashimi", 4)
end

if minetest.get_modpath("farming") and farming.mod == "redo" then
	register_food("farming:bread", 6)
	register_food("farming:potato", 1)
	register_food("farming:baked_potato", 6)
	register_food("farming:cucumber", 4)
	register_food("farming:tomato", 4)
	register_food("farming:carrot", 3)
	register_food("farming:carrot_gold", 6, "", nil, 8)
	register_food("farming:corn", 3)
	register_food("farming:corn_cob", 5)
	register_food("farming:melon_slice", 2)
	register_food("farming:pumpkin_slice", 1)
	register_food("farming:pumpkin_bread", 9)
	register_food("farming:coffee_cup", 2, "farming:drinking_cup")
	register_food("farming:coffee_cup_hot", 3, "farming:drinking_cup", nil, 2)
	register_food("farming:cookie", 2)
	register_food("farming:chocolate_dark", 3)
	register_food("farming:donut", 4)
	register_food("farming:donut_chocolate", 6)
	register_food("farming:donut_apple", 6)
	register_food("farming:raspberries", 1)
	register_food("farming:blueberries", 1)
	register_food("farming:muffin_blueberry", 4)
	if minetest.get_modpath("vessels") then
		register_food("farming:smoothie_raspberry", 2, "vessels:drinking_glass")
	end
	register_food("farming:rhubarb", 1)
	register_food("farming:rhubarb_pie", 6)
	register_food("farming:beans", 1)
end

if minetest.get_modpath("kpgmobs") ~= nil then
	register_food("kpgmobs:uley", 3)
	register_food("kpgmobs:meat", 6)
	register_food("kpgmobs:rat_cooked", 5)
	register_food("kpgmobs:med_cooked", 4)
  	if minetest.get_modpath("bucket") then
	   register_food("kpgmobs:bucket_milk", 4, "bucket:bucket_empty")
	end
end

if minetest.get_modpath("jkfarming") ~= nil then
	register_food("jkfarming:carrot", 3)
	register_food("jkfarming:corn", 3)
	register_food("jkfarming:melon_part", 2)
	register_food("jkfarming:cake", 3)
end

if minetest.get_modpath("jkanimals") ~= nil then
	register_food("jkanimals:meat", 6)
end

if minetest.get_modpath("jkwine") ~= nil then
	register_food("jkwine:grapes", 2)
	register_food("jkwine:winebottle", 1)
end

if minetest.get_modpath("cooking") ~= nil then
	register_food("cooking:meat_beef_cooked", 4)
	register_food("cooking:fish_bluewhite_cooked", 3)
	register_food("cooking:fish_clownfish_cooked", 1)
	register_food("cooking:meat_chicken_cooked", 2)
	register_food("cooking:meat_cooked", 2)
	register_food("cooking:meat_pork_cooked", 3)
	register_food("cooking:meat_toxic_cooked", -3)
	register_food("cooking:meat_venison_cooked", 3)
	register_food("cooking:meat_undead_cooked", 1)
end

-- ferns mod of plantlife_modpack
if minetest.get_modpath("ferns") ~= nil then
	register_food("ferns:fiddlehead", 1, "", 1)
	register_food("ferns:fiddlehead_roasted", 3)
	register_food("ferns:ferntuber_roasted", 3)
	register_food("ferns:horsetail_01", 1)
end

if minetest.get_modpath("pizza") ~= nil then
	register_food("pizza:pizza", 30, "", nil, 30)
	register_food("pizza:pizzaslice", 5, "", nil, 5)
end
